src/v/cluster/health_monitor_backend.cc

```diff
@@ -827,4 +827,28 @@ bool health_monitor_backend::does_raft0_have_leader() {
     return _raft0->get_leader_id().has_value();
 }
 
+void health_monitor_backend::reset() {
+    vlog(clusterlog.info, "Resetting Health Monitor Backend");
+    // Exlicitly stop/wipe state here instead of calling stop() because stop()
+    // will also close the gate and break the mutex
+    _raft_manager.local().unregister_leadership_notification(
```

Comment:
does it make sense to include this? I say this because although there is state within `_raft_manager` to make this notification work, it isn't anything that could become stale -- it never changes for the duration of the program. The state should boil down to a reference to this notification lambda instance.

Reply:
I'm asking the same question in https://github.com/redpanda-data/redpanda/pull/9282#discussion_r1130686006

Reply:
I'll remove it for now. If someone else thinks we should re-register the leadership notification then I'll put it back in.